ui/tests/helpers.ts

```diff
@@ -138,6 +139,29 @@ export async function verifyDashboardRoute(page: Page) {
   await expect(page).toHaveURL("/");
 }
 
+export async function authenticateAndSaveState(
+  page: Page,
+  email: string,
+  password: string,
+  storagePath: string,
+) {
+  if (!email || !password) {
+    throw new Error('Email and password are required for authentication and save state');
+  }
+
+  // Create SignInPage instance
+  const signInPage = new SignInPage(page);
+  const credentials: SignInCredentials = { email, password };
+
+  // Perform authentication steps using Page Object Model
+  await signInPage.goto();
+  await signInPage.login(credentials);
+  await signInPage.verifySuccessfulLogin();
+
+  // Save authentication state
+  await page.context().storageState({ path: storagePath });
+}
```

Comment:
There could be a race condition. After verifySuccessfulLogin(), we should ensure that all cookies/tokens have been fully set before saving the state. 

```suggestion
  // Perform authentication steps using Page Object Model
  await signInPage.goto();
  await signInPage.login(credentials);
  await signInPage.verifySuccessfulLogin();
  await page.waitForLoadState('networkidle');
  // Save authentication state
  await page.context().storageState({ path: storagePath });
}
```

Reply:
Thanks for letting me know. I'll put it in the verifyLoad of the pages themselves.